### PR TITLE
Add metadata, support for relasing, package checks

### PR DESCRIPTION
**Pull request by justusschock**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -41,17 +41,17 @@ repos:
       - id: docformatter
         args: [--in-place, --wrap-summaries=120, --wrap-descriptions=120]
 
-  # - repo: https://github.com/PyCQA/isort
-  #   rev: 5.10.1
-  #   hooks:
-  #     - id: isort
-  #       args: [--profile, black]
+  - repo: https://github.com/PyCQA/isort
```
Comment (lantiga) [inline .pre-commit-config.yaml:44]: can you make sure that we pre-commit hooks don't modify code in the nanogpt and mingpt directories?

**Review by lantiga — CHANGES_REQUESTED** [review]
Pre-commit hooks shouldn't impact minGPT and nanoGPT (once it lands).

I propose to merge the nanoGPT PR, and update this one with the extra folder.

File: mingpt/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from mingpt.__about__ import *  # noqa: F401, F403
```
Comment (justusschock) [inline mingpt/__init__.py:1]:
```suggestion
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (justusschock) [reply .pre-commit-config.yaml]: done in #28

**Review by lantiga — APPROVED** [review]
Looks good, however let's decide on the name for the package.
I'm leaning towards `lightning-GPT` at this point. WDYT?

**Pull request merged**